MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -314,21 +314,42 @@ namespace WPEFramework {
                 /* Network check */
                 internetConnectStatus = isDeviceOnline();
             }
-#else
+#else /* WhoAmI */
             internetConnectStatus = isDeviceOnline();
 #endif
 
 #if defined(ENABLE_WHOAMI)
     string activation_status = checkActivatedStatus();
+    bool whoAmIStatus = false;
     if (UNSOLICITED_MAINTENANCE == g_maintenance_type) {
         /* WhoAmI check*/
-        bool whoAmIStatus = knowWhoAmI(activation_status);
+        whoAmIStatus = knowWhoAmI();
         if (whoAmIStatus) {
             LOGINFO("knowWhoAmI() returned successfully");
         }
+        else {
+            LOGINFO("knowWhoAmI() returned false");
+        }
     }
-            if ( false == internetConnectStatus && activation_status == "activated" ) {
-#else
+
+    if (false == whoAmIStatus && activation_status != "activated") {
+        bool debugReturnFlag = false;
+        LOGINFO("knoWhoAmI() returned false and Device is not already Activated");
+        g_listen_to_deviceContextUpdate = true;
+        LOGINFO("Waiting for thread.. ");
+        task_thread.wait(lck);
+        LOGINFO("Resuming thread and Set Device Initialization Context Data from SecManager onDeviceInitializationContextUpdate event");
+        debugReturnFlag = setDeviceInitializationContext(g_jsonRespDeviceInitialization);
+        if (debugReturnFlag) {
+            LOGINFO("Device Initialization Context data set via Event Handler success");
```

Comment:
Lets keep it simple. LOG "setDeviceInitializationContext() success" or "setDeviceInitializationContext() failed"